### PR TITLE
New Rule: Suspicious Request for quote or purchase

### PR DESCRIPTION
**Pull request by morriscode**
I'm not married to the name

### CODE REVIEW

File: detection-rules/suspicious_request_for_quote_or_purchase.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          (
+              sender.email.domain.root_domain in $free_email_providers
+              and sender.email.email not in $recipient_emails
+          )
+          or 
+          (
+              sender.email.domain.root_domain not in $free_email_providers
+              and sender.email.domain.domain not in $recipient_domains
+          )
+      )
```
Comment (jkamdjou) [inline detection-rules/suspicious_request_for_quote_or_purchase.yml:33]: may cause FPs since it's 1 of

Comment (morriscode) [reply detection-rules/suspicious_request_for_quote_or_purchase.yml:33]: I think you're right... Pulling unsolicited increases accuracy

**Review by aidenmitchell — APPROVED** [review]

File: detection-rules/suspicious_request_for_quote_or_purchase.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            )
+            and length(recipients.cc) == 0
+            and length(recipients.bcc) == 0
+        ),
+        (
+            sender.email.domain.root_domain in $free_email_providers
+            and any(headers.reply_to, .email.email != sender.email.email)
+            and any(headers.reply_to, .email.email not in $recipient_emails)
+        )
+  )
```
Comment (jkamdjou) [inline detection-rules/suspicious_request_for_quote_or_purchase.yml:23]: this snippet is 🔥  just heads up that this doesn't mean there's necessarily a suspicious sender (in the rule name)

File: detection-rules/suspicious_request_for_quote_or_purchase.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+name: "Request for Quote or Purchase (RFQ|RFP) from suspicious sender"
```
Comment (jkamdjou) [inline detection-rules/suspicious_request_for_quote_or_purchase.yml:1]:
```suggestion
name: "Request for Quote or Purchase (RFQ|RFP) with suspicious sender or recipient pattern"
```

File: detection-rules/suspicious_request_for_quote_or_purchase.yml (hunk 2 above)
Comment (morriscode) [reply detection-rules/suspicious_request_for_quote_or_purchase.yml:23]: Love the name change ty ty ty!

**Comment by morriscode** [comment]
New batch of test show high accuracy and overlap with other confidence rules. This is GTG

**Comment by morriscode** [comment]
I think i'm seeing a null value in the test data for this, going to close the PR and reopen and run test-rules again.

**Comment by jkamdjou** [comment]
> I think i'm seeing a null value in the test data for this, going to close the PR and reopen and run test-rules again.

copy, lmk when you're ready for a re-review

**Comment by jkamdjou** [comment]
@morriscode it looks like it's been running: https://github.com/sublime-security/sublime-rules/blob/test-rules/detection-rules/suspicious_request_for_quote_or_purchase.yml

**Comment by morriscode** [comment]
> @morriscode it looks like it's been running: https://github.com/sublime-security/sublime-rules/blob/test-rules/detection-rules/suspicious_request_for_quote_or_purchase.yml

It's been running but returned 3 values for the sha. The original, a null which looks like it was getting populated more recently and a third sha with no hits. 

To err on the side of caution I just wanted to reinitiate it. I'm wonder if we close a PR but don't wait for the cleanup job to run before opening it if it actually removes it...

**Comment by morriscode** [comment]
I suspect if we close and reopen a PR without waiting for the remove stale rules job it won't recognize that the PR was close as it's a point in time state check... Gonna close this and test

**Comment by morriscode** [comment]
@jkamdjou I think this is gtg, but it's really hard to discern the test-rule-data in this scenario.

**Review by jkamdjou — APPROVED** [review]

**Pull request merged**